### PR TITLE
[New] Show extruded features

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Show extruded features` in `Scenes` category.
URL to README: https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-ShowExtrudedFeatures/Shared/Samples/Show%20extruded%20features

## Linked Issue(s)

- `swift/issues/4191`

## Screenshots

show-extruded-features

### CODE REVIEW

File: Shared/Samples/Show extruded features/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+1. Create a `ServiceFeatureTable` from a URL.
+2. Create a feature layer from the service feature table.
+   * Make sure to set the rendering mode to dynamic, `featureLayer.setRenderingMode = .dymanic`.
+3. Apply a `SimpleRenderer` to the feature layer.
+4. Set `ExtrusionMode` of render, `renderer.sceneProperties.extrusionMode = .absoluteHeight`.
+5. Set extrusion expression of renderer, `renderer.sceneProperties.extrusionExpression = "[POP2007]/ 10"`.
+
+## Relevant API
+
+* ExtrusionExpression
```
Comment (yo1995) [inline Shared/Samples/Show extruded features/README.md:26]: This doesn't seem to be an ArcGIS symbol

File: Shared/Samples/Show extruded features/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+* ExtrusionMode
```
Comment (yo1995) [inline Shared/Samples/Show extruded features/README.md:27]: RendererSceneProperties.ExtrusionMode?

File: Shared/Samples/Show extruded features/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+* FeatureLayer
+* SceneProperties
```
Comment (yo1995) [inline Shared/Samples/Show extruded features/README.md:29]: RendererSceneProperties

File: Shared/Samples/Show extruded features/ShowExtrudedFeaturesView.swift [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+struct ShowExtrudedFeaturesView: View {
+    /// A scene with a topographic basemap centered on the US.
+    @State private var scene: ArcGIS.Scene = {
+        let scene = Scene(basemapStyle: .arcGISTopographic)
+        
+        // Set the scene view's viewpoint specified by the camera position.
+        let point = Point(x: -99.659448, y: 20.513652, z: 12_940_924, spatialReference: .wgs84)
+        let camera = Camera(lookingAt: point, distance: 0, heading: 0, pitch: 15, roll: 0)
+        scene.initialViewpoint = Viewpoint(boundingGeometry: point, camera: camera)
```
Comment (yo1995) [inline Shared/Samples/Show extruded features/ShowExtrudedFeaturesView.swift:26]: Me personally don't like using a point as "bounding geometry" as it has no effect, and doesn't make sense grammatically. Instead, for setting a 3D viewpoint with a camera, use [Viewpoint.init(center:scale:rotation:camera:)](https://developers.arcgis.com/swift/api-reference/documentation/arcgis/viewpoint/init(center:scale:rotation:camera:))

File: Shared/Samples/Show extruded features/ShowExtrudedFeaturesView.swift [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    /// A enum for the different population statistics.
+    enum Statistic: Int {
+        case totalPopulation
+        case populationDensity
+        
+        /// The extrusion expression for the statistic.
+        var extrusionExpression: String {
+            switch self {
+            case .totalPopulation:
+                return "[POP2007]/ 10"
```
Comment (yo1995) [inline Shared/Samples/Show extruded features/ShowExtrudedFeaturesView.swift:105]:
```suggestion
                return "[POP2007] / 10"
```

File: Shared/Samples/Show extruded features/ShowExtrudedFeaturesView.swift [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                Text("Total Population").tag(Statistic.totalPopulation)
+                Text("Population Density").tag(Statistic.populationDensity)
+            }
+            .pickerStyle(.segmented)
+            .padding()
+            .onChange(of: statisticSelection) { _ in
+                if let statistic = Statistic(rawValue: statisticSelection.rawValue) {
+                    renderer.sceneProperties.extrusionExpression = statistic.extrusionExpression
+                }
+            }
```
Comment (yo1995) [inline Shared/Samples/Show extruded features/ShowExtrudedFeaturesView.swift:91]:
```suggestion
            .onChange(of: statisticSelection) { newValue in
                featureLayer.renderer?.sceneProperties.extrusionExpression = newValue.extrusionExpression
                }
            }
```
If the suggestions above are applied.

File: Shared/Samples/Show extruded features/README.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+3. Apply a `SimpleRenderer` to the feature layer.
+4. Set `ExtrusionMode` of render, `renderer.sceneProperties.extrusionMode = .absoluteHeight`.
+5. Set extrusion expression of renderer, `renderer.sceneProperties.extrusionExpression = "[POP2007] / 10"`.
+
+## Relevant API
+
+* FeatureLayer
+* RendererSceneProperties
+* RendererSceneProperties.ExtrusionMode
+* SceneProperties
```
Comment (yo1995) [inline Shared/Samples/Show extruded features/README.md:29]:
```suggestion
```

File: Shared/Samples/Show extruded features/show-extruded-features.png
Comment (yo1995) [inline Shared/Samples/Show extruded features/show-extruded-features.png:1]: Don't forget to update the screenshot.

**Comment by yo1995** [comment]
Shuffled Nimesh out as he is focusing on feature team work, no time to review PRs.

File: Shared/Samples/Show extruded features/ShowExtrudedFeaturesView.swift [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    private var featureLayer: FeatureLayer {
+        scene.operationalLayers.last as! FeatureLayer
+    }
+    
+    var body: some View {
+        VStack {
+            SceneView(scene: scene)
+            
+            // Unit system picker.
+            Picker("", selection: $statisticSelection) {
```
Comment (dfeinzimer) [inline Shared/Samples/Show extruded features/ShowExtrudedFeaturesView.swift:35]:
```suggestion
            Picker("Statistic", selection: $statisticSelection) {
```

Though the `titleKey` doesn't appear to be expressly used at present, it _may_ be used internally for accessibility in some way. I'm not certain either way.

Passing some name though, such as "Statistic" does make the sample code a bit clearer as to what the picker is for and would probably also allow you to remove the inline comment directly above.

**Review by dfeinzimer — DISMISSED** [review]
Just one suggestion. Take or leave it. Neat sample!

**Review by dfeinzimer — DISMISSED** [review]

File: Shared/Samples/Show extruded features/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+## How to use the sample
+
+Tap the bottom buttons to switch between using population density and total population for extrusion. Higher extrusion directly corresponds to higher attribute values.
+
+## How it works
+
+1. Create a `ServiceFeatureTable` from a URL.
+2. Create a feature layer from the service feature table.
+   * Make sure to set the rendering mode to dynamic, `featureLayer.setRenderingMode = .dymanic`.
```
Comment (yo1995) [inline Shared/Samples/Show extruded features/README.md:19]:
```suggestion
    * Make sure to set the rendering mode to dynamic, `featureLayer.setRenderingMode = .dynamic`.
```
Sorry I didn't catch earlier - 4 spaces for list indentation

**Review by yo1995 — APPROVED** [review]

**Review by dfeinzimer — APPROVED** [review]

**Pull request merged**